Only run lint checks against changed files

KaiSforza opened the pull request:
### What does this PR do?

Changes the behaviour of `pr-lint` to only check the files that git says changed between the merge target and the PR branch.

### What issues does this PR fix or reference?

Re-does #48610 

### Previous Behavior
Runs pylint on every file in the repo

### New Behavior
Only runs on changed files using find and git diff

### Tests written?

These are the tests.

### Commits signed with GPG?

Yes

@gtmanfred @dubb-b

KaiSforza committed:
Only run lint checks against changed files

gtmanfred reviewed (changes requested):
can you fix this pylint error in the testprogram.py file?

```
16:56:23 [test linting] ************* Module tests.integration.utils.testprogram
16:56:23 [test linting] tests/integration/utils/testprogram.py:601: [W0702(bare-except), TestSaltProgram] No exception type(s) specified
```

rallytime commented:
I don't think this is quite right - the lint job says it passed, but it has a warning in the list: https://jenkinsci.saltstack.com/job/pr-lint/job/PR-48783/1/

My next question is why is this running on that `testprogram` file when you didn't change that file in this PR? Might be jumping the gun on that, but I am curious about how this should be working.

KaiSforza commented:
@rallytime the warning/pass is a whole other issue though. Currently the run isn't using the updated pylint script, so it's running against everything.

KaiSforza committed:
test: except for OSError only

KaiSforza commented:
@rallytime also i think that if there are no changes to the files we're finding against, it'll run against everything, at least for the test run.

rallytime commented:
> the warning/pass is a whole other issue though.

I'm not quite following. What other issue? This is working in PRs as I would expect it to - we get failures when there are lint issues. See https://github.com/saltstack/salt/pull/48781 as an example. If this is changing those results, then that's not good.

KaiSforza committed:
Threshold was wrong

I also removed the delta value check. It's going to be weird with the
number of runs we do, and it's also set to run on failures.

KaiSforza commented:
@rallytime The failures happen if we have more than one issue, but with one it was below the threshold. I fixed it in a commit here.

rallytime commented:
@KaiSforza Perfect! Thank you :)
